src/App.jsx

```diff
@@ -95,12 +95,32 @@ function App() {
     // network. To actually test rinkeby, change setChainID equal to 4 before testing.
     let loadProvider = provider;
 
-    await dispatch(loadAppDetails({ networkID: chainID, provider: loadProvider }));
-    if (address) await dispatch(loadAccountDetails({ networkID: chainID, address, provider: loadProvider }));
-
-    Object.values(BONDS).map(async bond => {
-      await dispatch(calcBondDetails({ bond, value: null, provider: loadProvider, networkID: chainID }));
-    });
+    // NOTE (appleseed): loadDetails() runs three times on every app refresh...
+    // ... once with address === "" && loadProvider === StaticJsonRpcProvider (set inside of Web3ContextProvider)
+    // ... once with address === "[wallet address]" && loadProvider === StaticJsonRpcProvider (set inside of Web3Context.connect())
+    // ... once with address === "[wallet address]" && loadProvider === Web3Provider (set inside of Web3Context.connect() right after the above line)
+    // So we need to make sure we don't run `loadAccountDetails`, `loadAppDetails` & `calcBondDetails`...
+    // ... below each of the three times state is changed
+    // The below if statements are one way to prevent the 3x runs
+    //
+    // don't run except when address === "" && provider is a API Provider (not Metamask)
+    // `loadDetails()` always runs once with address === "" even when the user has a connected wallet.
+    if (address === "" && loadProvider.connection["url"] !== "metamask") {
```

Comment:
We can't assume the url is metamask, we also. support Walletconnect and potentially other wallet providers!

I added a convenience method `provider.connected` which should be `true` if `web3Context.connect()` succeeds.

Reply:
It looks like web3Context.connect() is true for both Wallet Providers and API providers, but I like this idea. I'm digging through your other methods to see what else I can use

Reply:
hmm.. it _shouldn't_ be true for api providers. lets fix that then. I think `connected` should denote when we're connected with a wallet. Thoughts?